pkg/infrastructure/services/resources.go

```diff
@@ -38,38 +40,74 @@ const (
 	}
 
-	service := createRequiredService(instance, deployment)
+	// we only create the rest of the resources once we have a resolvable image
+	// or if the deployment is already there, we don't want to delete it :)
+	if image, err := s.getKogitoServiceImage(imageHandler, instance); err != nil {
+		return resources, err
+	} else if len(image) > 0 {
+		deployment := createRequiredDeployment(instance, image, s.definition)
+		if err = s.applyDeploymentCustomizations(deployment, instance, imageHandler); err != nil {
+			return resources, err
+		}
+
+		service := createRequiredService(instance, deployment)
 
-	if s.definition.infinispanAware {
-		if err = s.applyInfinispanConfigurations(deployment, instance); err != nil {
-			return
+		if s.definition.infinispanAware {
+			if err = s.applyInfinispanConfigurations(deployment, instance); err != nil {
+				return resources, err
+			}
+		}
+		if s.definition.kafkaAware {
+			if err = s.applyKafkaConfigurations(deployment, instance); err != nil {
+				return resources, err
+			}
 		}
-	}
 
-	if s.definition.kafkaAware {
-		if err = s.applyKafkaConfigurations(deployment, instance); err != nil {
-			return
+		resources[reflect.TypeOf(appsv1.Deployment{})] = []resource.KubernetesResource{deployment}
+		resources[reflect.TypeOf(corev1.Service{})] = []resource.KubernetesResource{service}
+		if s.client.IsOpenshift() {
+			resources[reflect.TypeOf(routev1.Route{})] = []resource.KubernetesResource{createRequiredRoute(instance, service)}
 		}
 	}
 
-	resources[reflect.TypeOf(appsv1.Deployment{})] = []resource.KubernetesResource{deployment}
-	resources[reflect.TypeOf(corev1.Service{})] = []resource.KubernetesResource{service}
+	return
+}
 
+func (s *serviceDeployer) applyDeploymentCustomizations(deployment *appsv1.Deployment, instance v1alpha1.KogitoService, imageHandler *imageHandler) error {
 	if imageHandler.hasImageStream() {
-		resources[reflect.TypeOf(imgv1.ImageStream{})] = []resource.KubernetesResource{imageHandler.imageStream}
+		key, value := framework.ResolveImageStreamTriggerAnnotation(imageHandler.resolveImageNameTag(), instance.GetName())
+		deployment.Annotations = map[string]string{key: value}
 	}
-	if s.client.IsOpenshift() {
-		resources[reflect.TypeOf(routev1.Route{})] = []resource.KubernetesResource{createRequiredRoute(instance, service)}
+	if s.definition.OnDeploymentCreate != nil {
+		if err := s.definition.OnDeploymentCreate(deployment, instance); err != nil {
+			return err
+		}
 	}
+	return nil
+}
 
-	return
+func (s *serviceDeployer) getKogitoServiceImage(imageHandler *imageHandler, instance v1alpha1.KogitoService) (string, error) {
+	image, err := imageHandler.resolveImage()
+	if err != nil {
+		return "", err
+	}
+	if len(image) > 0 {
+		return image, nil
+	}
+	deploymentDeployed := &appsv1.Deployment{ObjectMeta: v1.ObjectMeta{Name: instance.GetName(), Namespace: instance.GetNamespace()}}
```

Comment:
Shouldn't DeploymentConfig be retrieved here?

Reply:
Or maybe DeploymentConfig in case of OpenShift, Deployment for Kubernetes?

Reply:
We're not using DeploymentConfig anymore for Kogito Services and I'll also remove it from KogitoApp in the future, when we break it into Runtimes and Build. Kogito Apps will use this same API.

Reply:
Oh, right, thanks.